### DISCUSSION TITLE
What is the best way to identify a sum of instances of a certain function in SymPy?

### DISCUSSION THREAD

**Discussion by evbernardes** [sympy/sympy#24591] [Q&A] [answered]
For example, let's say I have an expression equivalent to $t = \operatorname{atan_2}(y1, x1) + \operatorname{exp}(x1) - \operatorname{atan_2}(y2, x2)$.

Is there a function I can use to quickly return a list of all the `atan2` elements in this expression, taking into account the sign (since the negative `atan2` element would actually be stored as a `mul`)?

Moreover, after doing that, what is the preferred way to check it this returned sum is "simpler" than some equivalent simplification I would do to it?

**Comment by sylee957** [comment]
I think that it works `t.atoms(atan2)` would work, but I'm not sure it's the answer expected because I'm not sure about the part

> taking into account the sign (since the negative atan2 element would actually be stored as a mul)?

**Reply by evbernardes** [reply]
> I think that it works `t.atoms(atan2)` would work, but I'm not sure it's the answer expected because I'm not sure about the part

Thanks for the input! Sadly it doesn't take the multiplying constant indeed, so I'll have to try something else

**Comment by oscarbenjamin** [comment, marked as answer]
You can use `as_coefficients_dict`:
```python
In [17]: y1, x1, y2, x2 = symbols('y1, x1, y1, x2')

In [18]: t = atan2(y1, x1) + exp(x1) - atan2(y2, x2)

In [19]: print(t.as_coefficients_dict(*t.atoms(atan2)))
defaultdict(<class 'int'>, {atan2(y1, x2): -1, atan2(y1, x1): 1, 1: exp(x1)})
```

**Reply by evbernardes** [reply]
This one might help, thanks!
